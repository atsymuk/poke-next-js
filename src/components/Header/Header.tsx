import type { VFC } from 'react'
import Link from 'next/link'

import styles from './header.module.scss'

interface HeaderProps {
  backUrl?: string
  title: string
}

const Header: VFC<HeaderProps> = ({ backUrl, title }) => (
  <div className={styles.container}>
    {backUrl && (
      <Link href={backUrl}>
        <a className={styles.backLink}>{'< Back'}</a>
      </Link>
    )}
    <h1 className={styles.header}>{title}</h1>
  </div>
)

export default Header
