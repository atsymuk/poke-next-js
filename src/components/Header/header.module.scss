@import "styles/breakpoints.scss";
@import "styles/theme.scss";

.container {
  align-items: center;
  background-color: $primaryBackgroundColorDark;
  display: flex;
  justify-content: center;
  height: 300px;
  position: relative;
}

.header {
  color: $primaryTextColor;
  font-size: 30px;
  font-weight: 700;
  text-transform: capitalize;
}

.backLink {
  color: $primaryTextColor;
  font-size: 24px;
  left: 20px;
  position: absolute;
  text-decoration: underline;
  top: 20px;
}

@media (max-width: $sm) {
  .container {
    height: 150px;
  }

  .header {
    font-size: 20px;
  }

  .backLink {
    font-size: 18px;
  }
}
