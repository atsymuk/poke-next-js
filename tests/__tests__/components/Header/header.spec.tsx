import { render } from '@testing-library/react'
import Header from 'components/Header/Header'

it('Header component snapshot', () => {
  const { container } = render(<Header title="test" />)
  expect(container).toMatchSnapshot()
})
