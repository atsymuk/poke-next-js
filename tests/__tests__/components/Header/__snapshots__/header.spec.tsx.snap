// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1
      class="header"
    >
      test
    </h1>
  </div>
</div>
`;
